pkg/resource/feature_group/manager_test_suite_test.go

```diff
@@ -0,0 +1,144 @@
+	"go.uber.org/zap/zapcore"
+)
+
+// provideResourceManagerWithMockSDKAPI accepts MockSageMakerAPI and returns pointer to resourceManager
+// the returned resourceManager is configured to use mockapi api.
+func provideResourceManagerWithMockSDKAPI(mockSageMakerAPI *mocksvcsdkapi.SageMakerAPI) *resourceManager {
+	zapOptions := ctrlrtzap.Options{
+		Development: true,
+		Level:       zapcore.InfoLevel,
+	}
+	fakeLogger := ctrlrtzap.New(ctrlrtzap.UseFlagOptions(&zapOptions))
+	return &resourceManager{
+		rr:           nil,
+		awsAccountID: "",
+		awsRegion:    "",
+		sess:         nil,
+		sdkapi:       mockSageMakerAPI,
+		log:          fakeLogger,
+		metrics:      ackmetrics.NewMetrics("sagemaker"),
+	}
+}
+
+// TestDeclarativeTestSuite runs the test suite for feature group
+func TestDeclarativeTestSuite(t *testing.T) {
+     	defer func() {
+     	   if r := recover(); r != nil {
+     	      fmt.Println(testutil.RecoverPanicString, r)
+	      t.Fail()
+     	   }
+        }()
+	var ts = testutil.TestSuite{}
+	testutil.LoadFromFixture(filepath.Join("testdata", "test_suite.yaml"), &ts)
+	var delegate = testRunnerDelegate{t: t}
+	var runner = testutil.TestSuiteRunner{TestSuite: &ts, Delegate: &delegate}
+	runner.RunTests()
+}
+
+// testRunnerDelegate implements testutil.TestRunnerDelegate
+type testRunnerDelegate struct {
+	t *testing.T
+}
+
+func (d *testRunnerDelegate) ResourceDescriptor() acktypes.AWSResourceDescriptor {
+	return &resourceDescriptor{}
+}
+
+func (d *testRunnerDelegate) ResourceManager(mocksdkapi *mocksvcsdkapi.SageMakerAPI) acktypes.AWSResourceManager {
+	return provideResourceManagerWithMockSDKAPI(mocksdkapi)
+}
+
+func (d *testRunnerDelegate) GoTestRunner() *testing.T {
+	return d.t
+}
+
+func (d *testRunnerDelegate) EmptyServiceAPIOutput(apiName string) (interface{}, error) {
+	if apiName == "" {
+		return nil, errors.New("no API name specified")
+	}
+	//TODO: use reflection, template to auto generate this block/method.
+	switch apiName {
+	case "CreateFeatureGroup":
+		var output svcsdk.CreateFeatureGroupOutput
+		return &output, nil
+	case "CreateFeatureGroupRequest":
+		var output svcsdk.CreateFeatureGroupOutput
+	    	return &output, nil
+	case "CreateFeatureGroupWithContext":
+		var output svcsdk.CreateFeatureGroupOutput
+	    	return &output, nil
+	case "DeleteFeatureGroup":
+		var output svcsdk.DeleteFeatureGroupOutput
+		return &output, nil
+	case "DeleteFeatureGroupRequest":
+		var output svcsdk.DeleteFeatureGroupOutput
+		return &output, nil
+	case "DeleteFeatureGroupWithContext":
+		var output svcsdk.DeleteFeatureGroupOutput
+		return &output, nil
+	case "DescribeFeatureGroup":
+		var output svcsdk.DescribeFeatureGroupOutput
+		return &output, nil
+	case "DescribeFeatureGroupRequest":
+		var output svcsdk.DescribeFeatureGroupOutput
+		return &output, nil
+	case "DescribeFeatureGroupWithContext":
+		var output svcsdk.DescribeFeatureGroupOutput
+		return &output, nil
+	case "ListFeatureGroups":
```

Comment:
can the list APIs be removed? controller doesnt never use them

Reply:
Yes, I ended up removing these in PRs #68 and #76. I have now added the change here as well.